types/seamless-immutable/index.d.ts

```diff
@@ -108,7 +108,7 @@ declare namespace SeamlessImmutable {
         /** Custom implementation of the array functions, which return Immutable. */
         interface Overrides<T> {
             forEach(callbackfn: (value: Immutable<T>, index: number, array: Immutable<T[]>) => void, thisArg?: any): void;
-            map<TTarget>(mapFuction: (item: Immutable<T>) => TTarget): Immutable<TTarget[]>;
+            map<TTarget>(mapFuction: (item: Immutable<T>, index?: number, array?: Immutable<T[]>) => TTarget): Immutable<TTarget[]>;
```

Comment:
`index` and `array` should not be marked as optional parameters, as function callback parameters are already optional by default. If `index` and `array` are marked as optional then when we use them in the mapping-function their types will be `number | undefined` and `Immutable<T[]> | undefined`, which is not the behaviour we want.

To demonstrate the above behaviour, consider the `forEach` function, which allows us to  omit index and array, without needing to mark those parameters as optional:
```
arr.forEach((value) => console.log({value})
```